examples/hybrid_jobs/6_QNSPSA_optimizer_with_embedded_simulator/qnspsa_with_embedded_simulator.ipynb

```diff
@@ -0,0 +1,1360 @@
+    "\n",
+    "- First, I would like my benchmarking tests to be reproducible, which requires a systematic way to track the logs and the outputs. This is easily done with Braket Jobs, but with the notebook, I would run into connection/permission timeout issues when the execution time of my program goes up to a few hours. I could easily lose the returned results or printed logs due to this issue. \n",
+    "\n",
+    "- Second, for tests on a larger scale with larger quantum circuits, I will likely need larger instances, with more abundant memory and computing power. For those more expensive instances, to be frugal, I would like to be charge only for the hours when my code is running. I would not be able to turn off the notebook instance at 3 am right after its execution, but Braket Jobs automatically do so to the Job instances."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2e1c9006",
+   "metadata": {},
+   "source": [
+    "Yet I will start with creating a working sample in the Braket Jobs local mode (`LocalQuantumJob`), instead of directly proceeding to execute it remotely (`AwsQuantumJob`). Braket service needs to prepare the environment (Docker container) for each remote Job I submit, and this process typically takes up to a few minutes. On the other hand, with the local mode, the environment only needs to be prepared once during my first call. I can fail fast and iterate with the local mode, and move to the remote Jobs after debugging to better utilize its power."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "c40facbf",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "292282985366.dkr.ecr.us-east-1.amazonaws.com/amazon-braket-pytorch-jobs:1.9.1-gpu-py38-cu111-ubuntu20.04\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Retrieve one of the default images (environment), with pennylane pre-installed\n",
+    "region_name = \"us-east-1\"\n",
+    "braket_client = boto3.client(\"braket\", region_name=region_name)\n",
+    "aws_session = AwsSession(braket_client=braket_client)\n",
+    "image_uri = retrieve_image(Framework.PL_PYTORCH, region_name)\n",
+    "print(image_uri)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "41e4154c",
+   "metadata": {},
+   "source": [
+    "Script `source_scripts/benchmark_qaoa_converge_speed` tests the performance of the four optimizers with the QAOA maxcut problem described above. As Pennylane does not provide a directly implementation on SPSA, here my SPSA optimzer is created by disabling the Fisher tensor metric estimation in QN-SPSA. In the script, I have adjusted the learning rates to optimize each of the optimizers' performance (tested separately). Considering the stochastic nature of SPSA and QN-SPSA, they are each repeated `spsa_repeats` times to capture the average performance. The below cell confirms that the Job works."
```

Comment:
"directly implementation" -> "direct implementation"

Reply:
"here my SPSA optimzer is created by disabling the Fisher tensor metric estimation in QN-SPSA": Let's rewrite this in active voice: "By disabling the Fisher tensor metric estimation in QN-SPSA, I created an SPSA optimizer."

Reply:
Also, it would be good to add one sentence about what the problem graph in `source_scripts/benchmark_qaoa_converge_speed` is.